src/core/qgsfielduiproperties.h

```diff
@@ -0,0 +1,59 @@
+#ifndef QGSFIELDUIPROPERTIES_H
+#define QGSFIELDUIPROPERTIES_H
+
+#include <QFont>
+#include <QColor>
+#include <QPixmap>
+
+#include "qgsfeature.h"
+
+/** \class QgsFieldFormat
+ * Hold formatting rules and style information for a field
+ */
+class CORE_EXPORT QgsConditionalStyle

```

Comment:
do you mind splitting off QgsConditionalStyle into its own header/cpp file? I'd really like to reuse this stuff in composer attribute tables, but in that case it's not based off a QgsField so QgsFieldUIProperties won't be used.